src/common/keygen.rs

```diff
@@ -0,0 +1,84 @@
+//! Contains functions used to generate scrambled alphabets from a key.
+//!
+
+use std::ascii::AsciiExt;
+use super::alphabet::{find_position, get_letter};
+
+/// Generates a scrambled alphabet using a key. Repeated letters in the key are ignored.
+/// e.g. A key of `alphabet` will produce the result `alphbetcdfgijkmnoqrsuvwxyz`.
+///
+/// Will return Err if a non-alphabetic symbol is present in the key.
+pub fn generate_keyed_alphabet(key: &str, make_uppercase: bool) -> Result<String, &'static str> {

```

Comment:
perhaps rename this to `keyed_alphabet` and callers would invoke it as `keygen::keyed_alphabet`